dc/s2n-quic-dc/events/map.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#[event("path_secret_map:initialized")]
+#[subject(endpoint)]
+struct PathSecretMapInitialized {
+    /// The capacity of the path secret map
+    capacity: usize,
+
+    /// The port that the path secret is listening on
+    control_socket_port: u16,
```

Comment:
I don't think we're listening on this port? Or at least we never read from the control socket -- it's purely used for sending. We listen/read control packets from the s2n-quic handshake socket.

It would be nice to have a *global* counter here for # of maps in existence -- probably this is the right place to emit that counter? (tracked via a global atomic static under the hood)

Reply:
> I don't think we're listening on this port? Or at least we never read from the control socket -- it's purely used for sending. We listen/read control packets from the s2n-quic handshake socket.

Do you think I should just remove it? I guess if we're not listening it's pretty useless.

> It would be nice to have a global counter here for # of maps in existence -- probably this is the right place to emit that counter? (tracked via a global atomic static under the hood)

I'm thinking the subscriber would be the one that tracks the number. Usually we try to keep the event emissions decoupled from the aggregation.

Reply:
Yeah I'd just drop the control socket port.

> I'm thinking the subscriber would be the one that tracks the number. Usually we try to keep the event emissions decoupled from the aggregation.

How can the subscriber keep track of a global number? We inherently need a global *in the map* if we want an actually global number.

Reply:
No, because I might not always configure the subscriber. The goal of the global static is to ensure that if at least one subscriber is used I can get an accurate count.

Maybe there's other ways to achieve that though.